qadence/noise/protocols.py

```diff
@@ -33,27 +66,161 @@ def get_noise_fn(self) -> Callable:
         return cast(Callable, fn)
 
     def _to_dict(self) -> dict:
-        return {"protocol": self.protocol, "options": self.options}
+        return {
+            "protocol": self.protocol,
+            "options": self.options,
+            "protocol_type": self.protocol_type,
+        }
 
     @classmethod
-    def _from_dict(cls, d: dict) -> Noise | None:
+    def _from_dict(cls, d: dict | None) -> NoiseSource | None:
         if d:
-            return cls(d["protocol"], **d["options"])
+            protocol_type = d.get("protocol_type", "")
+            return cls(d["protocol"], **d["options"], protocol_type=protocol_type)
         return None
 
     @classmethod
     def list(cls) -> list:
         return list(filter(lambda el: not el.startswith("__"), dir(cls)))
 
+    def __repr__(self) -> str:
+        return f"{self.protocol_type} NoiseSource({self.protocol}, {str(self.options)})"
+
+
+class NoiseHandler:
+    """A container for multiple sources of noise."""
+
+    def __init__(
+        self,
+        protocol: str | NoiseSource | list[str] | list[NoiseSource],
+        options: dict | list[dict] = dict(),
+        protocol_type: str | list[str] = "",
+    ) -> None:
+        self.noise_sources: list = list()
+        if isinstance(protocol, list) and isinstance(protocol[0], NoiseSource):
+            self.noise_sources += protocol
+        elif isinstance(protocol, NoiseSource):
+            self.noise_sources += [protocol]
+        else:
+            protocol = protocol if isinstance(protocol, list) else [protocol]
+            options = options if isinstance(options, list) else [options] * len(protocol)
+            types = (
+                protocol_type
+                if isinstance(protocol_type, list)
+                else [protocol_type] * len(protocol)
+            )
+
+            if len(options) != len(protocol) or len(types) != len(protocol):
```

Comment:
There might be noises that do not require options ? In that case, you'll fall in the first of you predicate cases without necessarily being wrong ?

Reply:
I am forcing noise without options to be an empty dict